### PR TITLE
Also record `next_batch` on `/sync` response

### PR DESCRIPTION
**Pull request by erikjohnston**
In the same vein as #5312.

### CODE REVIEW

File: changelog.d/5318.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Log the `since` token used and `next_batch` token returned when doing an incremental sync.
```
Comment (bmarty) [inline changelog.d/5318.misc:1]: Only updating the other file is enough, but do not worry I will delete this one later.

**Review by bmarty — APPROVED** [review]
LGTM

File: changelog.d/5318.misc (hunk 1 above)
Comment (erikjohnston) [reply changelog.d/5318.misc:1]: Ah, I figured I may as well combine the two PRs into a changelog entry (as they really should have been one PR), but YMMV :)

Comment (erikjohnston) [reply changelog.d/5318.misc:1]: (FTR towncrier merges multiple changelog files with the same content into a single line in the final changelog)

**Pull request merged**